multiuser/permission/che-multiuser-permission-workspace/src/main/java/org/eclipse/che/multiuser/permission/workspace/server/spi/jpa/MultiuserJpaWorkspaceDao.java

```diff
@@ -311,7 +316,35 @@ public void onCascadeEvent(BeforeAccountRemovedEvent event) throws Exception {
               (maxItems, skipCount) ->
                   workspaceManager.getByNamespace(
                       event.getAccount().getName(), false, maxItems, skipCount))) {
-        workspaceManager.removeWorkspace(workspace.getId());
+        // Will try to stop running/starting workspaces.
+        // At the moment is impossible to force stop for  STOPPING workspace
+        if (RUNNING.equals(workspace.getStatus()) || STARTING.equals(workspace.getStatus())) {
+          String owner = workspace.getRuntime().getOwner();
+          if (owner.equals(EnvironmentContext.getCurrent().getSubject().getUserId())) {
+            workspaceManager.stopWorkspace(
+                workspace.getId(), of(REMOVE_WORKSPACE_AFTER_STOP, "true"));
```

Comment:
Can we extract this to the private method and add a javadoc (it is also used in the `tryStopWorkspaceWithSA`)

Reply:
no. pls, don't add more private methods.

Reply:
done